projects/angular/src/data/datagrid/utils/key-navigation-utils.ts

```diff
@@ -42,8 +42,16 @@ export class KeyNavigationUtils {
     return coordinates;
   }
 
+  get averageRowHeight() {
+    const heightSum = Array.from(this.rows.values()).reduce((initial, row) => {
+      return initial + row.clientHeight;
+    }, 0);
+
+    return Math.round(heightSum / this.rows.length);
+  }
```

Comment:
We can use `requestAnimationFrame` to optimise it even better so we can get the clientHeights in a batch so we don't have to trigger reflows for each clientHeight.
```suggestion
  get averageRowHeight() {
      const rowHeights = [];
      requestAnimationFrame(() => {
          for (const row of this.rows.values()) {
            rowHeights.push(row.clientHeight);
          }
      });
      const sum = rowHeights.reduce((total, height) => total + height, 0);
      return Math.round(sum / rowHeights.length);
  }
```

Reply:
```suggestion
  get averageRowHeight() {
    const heightsSum = Array.from(this.rows.values()).reduce((sum, row) => {
      return sum + row.clientHeight;
    }, 0);

    return Math.round(sum / this.rows.length);
  }
```